connection_test.go

```diff
@@ -575,156 +554,107 @@ func executeQueryAndConfirmMessage(db *sql.DB, query string, expectedErrorTable
+		if err == nil {
+			t.Fatalf("should have raised an error")
+		}
+		_, ok := err.(*SnowflakeError)
+		if !ok {
+			t.Fatalf("should be snowflake error. err: %v", err)
+		}
+	})
 }
 
 func TestQueryContextError(t *testing.T) {
-	ctx := context.Background()
-	config, err := ParseDSN(dsn)
-	if err != nil {
-		t.Error(err)
-	}
-	sc, err := buildSnowflakeConn(ctx, *config)
-	if err != nil {
-		t.Error(err)
-	}
-	if err = authenticateWithConfig(sc); err != nil {
-		t.Error(err)
-	}
-
-	sr := &snowflakeRestful{
-		FuncPostQuery:  postQueryTest,
-		TokenAccessor:  getSimpleTokenAccessor(),
-		RequestTimeout: 10,
-	}
-
-	sc.rest = sr
+	runSnowflakeConnTest(t, func(sct *SCTest) {
+		ctx := sct.sc.ctx
+		sr := &snowflakeRestful{
+			FuncPostQuery:    postQueryTest,
+			FuncCloseSession: closeSessionMock,
+			TokenAccessor:    getSimpleTokenAccessor(),
+			RequestTimeout:   10,
+		}
 
-	_, err = sc.QueryContext(ctx, "SELECT 1", []driver.NamedValue{})
-	if err == nil {
-		t.Fatalf("should have raised an error")
-	}
+		sct.sc.rest = sr
+		_, err := sct.sc.QueryContext(ctx, "SELECT 1", []driver.NamedValue{})
+		if err == nil {
+			t.Fatalf("should have raised an error")
+		}
 
-	sc.rest.FuncPostQuery = postQueryFail
-	_, err = sc.QueryContext(ctx, "SELECT 1", []driver.NamedValue{})
-	if err == nil {
-		t.Fatalf("should have raised an error")
-	}
-	_, ok := err.(*SnowflakeError)
-	if !ok {
-		t.Fatalf("should be snowflake error. err: %v", err)
-	}
+		sct.sc.rest.FuncPostQuery = postQueryFail
+		_, err = sct.sc.QueryContext(ctx, "SELECT 1", []driver.NamedValue{})
+		if err == nil {
+			t.Fatalf("should have raised an error")
+		}
+		_, ok := err.(*SnowflakeError)
+		if !ok {
+			t.Fatalf("should be snowflake error. err: %v", err)
+		}
+	})
 }
 
 func TestPrepareQuery(t *testing.T) {
-	ctx := context.Background()
-	config, err := ParseDSN(dsn)
-	if err != nil {
-		t.Error(err)
-	}
-	sc, err := buildSnowflakeConn(ctx, *config)
-	if err != nil {
-		t.Error(err)
-	}
-	if err = authenticateWithConfig(sc); err != nil {
-		t.Error(err)
-	}
-	_, err = sc.Prepare("SELECT 1")
+	runSnowflakeConnTest(t, func(sct *SCTest) {
+		_, err := sct.sc.Prepare("SELECT 1")
 
-	if err != nil {
-		t.Fatalf("failed to prepare query. err: %v", err)
-	}
-	sc.Close()
+		if err != nil {
+			t.Fatalf("failed to prepare query. err: %v", err)
+		}
+		sct.sc.Close()
+	})
 }
 
 func TestBeginCreatesTransaction(t *testing.T) {
-	ctx := context.Background()
-	config, err := ParseDSN(dsn)
-	if err != nil {
-		t.Error(err)
-	}
-	sc, err := buildSnowflakeConn(ctx, *config)
-	if err != nil {
-		t.Error(err)
-	}
-	if err = authenticateWithConfig(sc); err != nil {
-		t.Error(err)
-	}
-	tx, _ := sc.Begin()
-	if tx == nil {
-		t.Fatal("should have created a transaction with connection")
-	}
-	sc.Close()
+	runSnowflakeConnTest(t, func(sct *SCTest) {
+		tx, _ := sct.sc.Begin()
+		if tx == nil {
+			t.Fatal("should have created a transaction with connection")
+		}
+		sct.sc.Close()
```

Comment:
We don't need it, `runSnowflakeConnTest` will close sc.

Reply:
Ok